src/plot_utils.py

```diff
@@ -86,12 +86,9 @@ def heatmap(bdb, bql, selectors=None, **kwargs):
     bdb : bayeslite.BayesDB
         Active BayesDB instance.
     bql : str
-        The BQL to run and plot. Must be a PAIRWISE BQL query.
-    selectors : [lambda name --> bool]
-        Rather than plot the full NxN matrix all together, make separate plots
-        for each combination of these selectors, plotting them in sequence.
-        If selectors are specified, returns a list of clustermaps, having
-        shown and closed each one.
+        The BQL to run and plot. Must be a PAIRWISE BQL query if specified.
+    df : pandas.DataFrame(columns=['generator_id', 'name0', 'name1', 'value'])
+        If bql is not specified, take data from here.
```

Comment:
I suggest raising an exception if both are specified in order to detect obvious mistakes.